### PR TITLE
Improve Serilog input

### PR DESCRIPTION
**Pull request by karolz-ms**
1. Added documentation for Serilog input
2. Added some unit tests
3. Improved value extraction logic to get better type fidelity between what gets logged vs. what EventFlow pipeline is seeing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -199,7 +199,70 @@ EventFlow PerformanceCounter input supports the first method of deterimining cou
...
+#### Serilog
+
+*Nuget package:* **Microsoft.Diagnostics.EventFlow.Inputs.Serilog**
+
+This input enables capturing diagnostic data created through [Serilog library](https://serilog.net/). It is designed to work with [Observable Serilog sink](https://github.com/serilog/serilog-sinks-observable). 
+
+*Configuration example*
+The Serilog input has no configuration, other than the "type" property that specifies the type of the input (must be "Serilog"):
+```json
+"inputs": [
```
Comment (yantang-msft) [inline README.md]: I think the "inputs" [] part is not needed, which is consistent with other inputs/outputs. User should know to put the config in the inputs section. And the below config example is not needed as well.

File: README.md [hunk 2]
```diff
@@ -199,7 +199,70 @@ EventFlow PerformanceCounter input supports the first method of deterimining cou
...
+namespace SerilogEventFlow
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            using (var pipeline = DiagnosticPipelineFactory.CreatePipeline(".\\eventFlowConfig.json"))
+            {
+                IObserver<LogEvent> serilogInput = pipeline.Inputs.OfType<SerilogInput>().First();
+                Log.Logger = new LoggerConfiguration().WriteTo.Observers(events => events.Subscribe(serilogInput)).CreateLogger();
```
Comment (yantang-msft) [inline README.md]: Don't know the details about Serilog. But should the Log.Logger be created by the Serilog input?
Ideally, user should just need one line of initialization and call Log.Info() however they want, just like the Trace input.

File: src/Microsoft.Diagnostics.EventFlow.Inputs.Serilog/SerilogInput.cs [hunk 3]
```diff
@@ -106,5 +106,42 @@ private EventData ToEventData(LogEvent logEvent)
...
+                    .Where(kvPair => kvPair.Key.Value is string && kvPair.Value is ScalarValue)
+                    .ToDictionary(kvPair => (string) kvPair.Key.Value, kvPair => ((ScalarValue) kvPair.Value).Value);
+                if (dictionaryResult.Count == dictionaryValue.Elements.Count)
+                {
+                    return dictionaryResult;
+                }
+            }
+
+            // Fall back to string rendering of the value
+            return logEventValue.ToString();
```
Comment (yantang-msft) [inline src/Microsoft.Diagnostics.EventFlow.Inputs.Serilog/SerilogInput.cs]: nit: Just curious, is ToString() overridden for StructureValue?

Comment (karolz-ms) [inline src/Microsoft.Diagnostics.EventFlow.Inputs.Serilog/SerilogInput.cs]: Yes, ToString() calls Render() which is overriden by StructureValue

File: README.md (hunk 2 above)
Comment (karolz-ms) [inline README.md]: I do not think we should be in the business of creating Serilog loggers. They have a lot of different options and a specific, fluent-style API to handle them--we do not want to duplicate that. Also and perhaps more importantly, it is expected to create many context-specific "child" loggers during the execution of a service. So it is not like EventFlow pipeline that typically gets created at startup and exists till the service is shut down

File: README.md (hunk 1 above)
Comment (karolz-ms) [inline README.md]: OK, I have removed the pieces you pointed out

File: README.md (hunk 2 above)
Comment (yantang-msft) [inline README.md]: OK, if it's necessary to create child loggers on the fly, then Serilog can be a special case given its complexity.

**Pull request merged**